app/responses/models.py

```diff
@@ -0,0 +1,95 @@
+from django.db import models
+
+from model_utils.managers import InheritanceManager
+
+from app.feedback_forms.models import FeedbackForm
+from app.prompts.models import (
+    BinaryPrompt,
+    Prompt,
+    RangedPrompt,
+    RangedPromptOption,
+    TextPrompt,
+)
+from app.utils.models import TimestampedModelMixin, UUIDModelMixin
+
+
+class Response(TimestampedModelMixin, UUIDModelMixin):
+    feedback_form = models.ForeignKey(
+        FeedbackForm, on_delete=models.PROTECT, related_name="+"
+    )
+    url = models.TextField()
+    metadata = models.JSONField()
+
+    def __str__(self):
+        return self.url
+
+
+class PromptResponse(TimestampedModelMixin, UUIDModelMixin):
+    objects = InheritanceManager()
+    prompt_type = Prompt
+
+    response = models.ForeignKey(
+        Response, on_delete=models.PROTECT, related_name="prompt_responses"
+    )
+    prompt = models.ForeignKey(
+        Prompt, on_delete=models.PROTECT, related_name="+"
+    )
+
+    def get_subclass_prompt(self) -> Prompt:
+        """
+        Gets the subclassed prompt based on the subclassed prompt response
+        """
+        if not isinstance(self.prompt, self.prompt_type):
```

Comment:
Could we add a comment to clarify that this checks if the prompt is not a concrete type but a base type, and we want to watch the concrete type if that's not the case? I think this may not be immediately obvious (well, took me 20 seconds).